service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Oskari(OgcApiSearchChannel.ID)
+public class OgcApiSearchChannel extends SearchChannel {
+
+
+    public static final String ID = "OGC_API_SEARCH_CHANNEL";
+
+    //Local configs
+    /**
+     * One can use property search.channel.OGC_API_SEARCH_CHANNEL.apikey but if really needed apikey can go here.
+     */
+    private static String apiKey = null;
+    private static String searchAddress = "https://avoin-paikkatieto.maanmittauslaitos.fi/geographic-names/features/v1/collections/placenames/items?spelling_case_insensitive=";
+    private static String additionalPlaceTypeInfoBaseUrl = "https://beta-paikkatieto.maanmittauslaitos.fi/catalogue/v1/codelistgroups/geographic-names/codelists/";
+    private static String[] localAdditionalInfoNames = {"placeType", "municipality"};
+    /**
+     * Set true if one wants to use local jsons instead of url based one
+     */
+    private static final boolean USE_LOCAL_ADDITIONAL_INFO = true;
+    /**
+     * Add languages you have translaited here as language code
+     */
+    private static String[] languageArray = {"fin", "eng", "swe"};
+
+    private static HashMap<Integer, HashMap<String, String>> placetypeCodes;
+    private static HashMap<Integer, HashMap<String, String>> municipalityCodes;
+
+    private Logger log = LogFactory.getLogger(this.getClass());
+
+    private static String CRS = null;
+    private static final String defaultCRS = "http://www.opengis.net/def/crs/EPSG/0/3067";
+
+
+    public ChannelSearchResult doSearch(SearchCriteria criteria) {
+        ChannelSearchResult result = new ChannelSearchResult();
+
+        if (placetypeCodes == null) {
+            placetypeCodes = buildAdditionalInfo(localAdditionalInfoNames[0]);
+        }
+        if (municipalityCodes == null) {
+            municipalityCodes = buildAdditionalInfo(localAdditionalInfoNames[1]);
+        }
+
+        //ApiKey Things
+        String propertyKey = PropertyUtil.get("search.channel." + ID + ".apikey");
+        if (propertyKey != null) {
+            String encodedApiKey = IOHelper.encode64(propertyKey + ":");
+            apiKey = "Basic " + encodedApiKey;
+
+        }
+        if (apiKey == null) {
+            log.error("OGC Api key was not found. OGC api will not return results.");
+        }
+        Map<String, String> headers = new HashMap<>(5);
+        headers.put("Authorization", apiKey);
+
+        String encodedSearchTerm = "";
+        try {
+            encodedSearchTerm = URLEncoder.encode(criteria.getSearchString(), StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+
+        String searchQuerry = searchAddress + encodedSearchTerm + "&limit=" + criteria.getMaxResults();
```

Comment:
Remove spelling_case_insensitive from base url and construct the url like this:
```
        Map<String, String> params = new LinkedHashMap<>();
        params.put("spelling_case_insensitive",criteria.getSearchString());
        params.put("limit", Integer.toString(criteria.getMaxResults() + 1));
        String url = IOHelper.constructUrl(searchAddress, params);
```
LinkedHashMap only makes it easier to write junit test for url construction but you could use HashMap() instead. If you query one more than the max results the framework can setup a flag to signal frontend that there would have been more results.